src/Components/Header.js

```diff
@@ -0,0 +1,87 @@
+'use strict'
+
+import React, {
+  Component,
+  PropTypes
+} from 'react'
+import AppBar from 'material-ui/lib/app-bar'
+import FlatButton from 'material-ui/lib/flat-button'
+import IconButton from 'material-ui/lib/icon-button'
+import NavigationClose from 'material-ui/lib/svg-icons/navigation/close'
+
+export default class Header extends Component {
+  constructor () {
+    super()
+    this._handleDeleteTap = this._handleDeleteTap.bind(this)
+  }
+
+  _renderRightElement (selectCount) {
+    return selectCount > 0
+      ? <FlatButton
+        labelStyle={{color: this.props.color}}
+        onTouchTap={this._handleDeleteTap}
+        label='Delete'
+        />
+          : <FlatButton
+            labelStyle={{color: this.props.color}}
+            onTouchTap={this.props.togglePermission}
+            label='Toggle Permission'
+            />
```

Comment:
Did you consider using the same element, but just conditionally changing the `label` and `onTouchTap`?